### ISSUE TITLE
Issues during reproduction

### ISSUE THREAD

**Issue by akriese** [MichaelWangGo/StereoMamba#1]
Hi, I am currently running (pretraining and finetuning) your architecture and have a couple of questions that I hoped you could answer for me.

1. From the instructions and the filename text files in `./datasets/filenames/` I was able to reproduce the pretraining on the sceneflow dataset. Unfortunately, the paper does not say a lot about the finetuning dataset choices. From `./datasets/filenames/scared_training.txt` I guess that the finetuning happens only on one frame per keyframe. Is that to avoid overfitting as the frames in a keyframe are too similar? As to why you're only picking datasets 1, 2, 3, 6 and 7, I guess you went by the conclusion from the MSDESIS paper about the other datasets being unusable for training, right?
2. During finetuning, I get some okayish validation Bad3-errors (between 9 and 12%), but when running the evaluation with the `./datasets/filenames/scared_testset.txt`, I get much worse values, which are also below the test values you state in the paper:

|source |EPE|Bad1|Bad2|Bad3|Bad5|Depth_MAE|FPS|
|---|---------|---------|---------|---------|---------|---------|---------|
|paper|2.64|xxxx|0.4149|0.2699|0.1388|2.55|21.28|
|repro|2.8763|0.6794|0.45529|0.30959|0.1663|2.903|22.22|

I guess, the good validation errors come from the fact, that the first frames of a keyframe are usually the most stable per keyframe. Have you made similar observations during finetuning and testing? What do you think about the test statistics of my reproduction. Are they within an expectable margin off your test values?

3. I am also trying to evaluate the performance of my reproduced model on StereoMIS, but somehow I couldnt find out how you came up with the data selection in `datasets/filenames/stereomis_testing.txt`. I followed the instructions in https://github.com/aimi-lab/robust-pose-estimator to extract the image files from the videos. In your filenames text file the paths look something like `/workspace/dataset/StereoMIS/P1/left_finalpass/frame_00000.png` which does not look like the file structure created by the paper's preprocessing. How did you prepare the dataset and how did you select the frames for testing? Neither the paper nor the repo seem to contain information on those questions. Over all frames of the P1 from StereoMIS, I am getting pretty bad SSIM (avg. 0.7242 vs your 0.9149) and PSNR (avg. 13.6 vs your 17.3) values. I'd like to be able to compare my checkpoint with yours for the exact same data selection.

I am sorry for the long text. I hope, you can take a bit of time to answer my questions and looking forward to your reply.

Thanks in advance and thank you for making your architecture openly accessible!

**Comment by MichaelWangGo** [comment]
Hi @akriese ,

1. We only used the first frame of each keyframe video is because only the first frame is annotated with disparity when they built the dataset, and the disparity maps of subsequent frames are calculated according to kinematcis. I think only the first frames are precise, while others could have errors when calculating. And picking datasets 1, 2, 3, 6 and 7 is because datasets 4 and 5 have calibration errors. Please refer to [the challenge paper](https://arxiv.org/abs/2101.01133)
2. Yes you are right, only the first frame is annotated. When using either StereoMamba or the comparison methods on the test set, I observed that the error per frame increases with the number of frames. However, theoretically, the error should fluctuate around a fixed value. Therefore, I believe that additional errors (such as kinematic errors) were introduced during the annotation of other frames. The values you've reproduced differ significantly from ours, which is unusual. We'll investigate shortly to determine whether this stems from a code version issue or another problem.
3. For preprocessing StereoMIS, I submitted the code, please refer to [here](https://github.com/MichaelWangGo/StereoMamba/blob/main/stereomis_video_2_calibrated_frames.py). And the original StereoMIS dataset should look like this:
<img width="238" height="324" alt="Image" src="https://github.com/user-attachments/assets/de4e889a-5b11-4bc8-9c56-f1947b217150" />

**Comment by akriese** [comment]
Thanks for quick and throrugh reply. I'll try to reproduce the test metrics with the code you've pushed.
Also thanks for the explanation about SCARED. I missed the fact with the groundtruth somehow.

I'll let you know, when I have some results

**Comment by akriese** [comment]
Judging from the amount of frames in the `stereomis_testing.txt` file, you picked a `frame_interval` of 10 in the preprocess-script
```py
process_video_and_extract(video_path, input_folder, output_folder, frame_interval=2)
```
Can you quickly confirm this?
If that is the case, I can continue creating and evaluating on the subset that you used for the paper's statitics.

**Comment by MichaelWangGo** [comment]
> Judging from the amount of frames in the `stereomis_testing.txt` file, you picked a `frame_interval` of 10 in the preprocess-script
> 
> process_video_and_extract(video_path, input_folder, output_folder, frame_interval=2)
> Can you quickly confirm this? If that is the case, I can continue creating and evaluating on the subset that you used for the paper's statitics.

Hi, 

Sorry I don't remember the frame_interval to be honest, maybe 8 or 10. but every video of stereomis is very long, so even if you select frame_interval of 10, you would get over 1k frames of each video, I don't think that would influence statistic results significantly.

**Comment by akriese** [comment]
Alright, I tested my checkpoint on the subset, which should be equal to the one you used. The SSIM is at 0.9139 which is almost equal to the paper's (0.9149). So that's a good thing :)
The PSNR is a little higher (21.12) than what the paper states (17.3), but that might stem from the fact that I ran the metric on values in the range [0, 1] while your code transforms the pixels to [-1, 1]. So, they are not really comparable, I think, and as running the evaluation took 2h, I wont run it with the other transformation.

I am still curious as to how the SCARED test metrics are so different while the StereoMIS oneshot results are somewhat similar. Maybe, I made a mistake in the SCARED testing. I'll look into it once more :)

**Comment by MichaelWangGo** [comment]
@akriese Not sure which checkpoint did you use. Here is our wandb record when finetuning on scared dataset. The final checkpoint could be overfitted, as you can see below. So you should find the checkpoint with the best performance according to the validation curve instead of the final one.

<img width="1006" height="307" alt="Image" src="https://github.com/user-attachments/assets/1a367fa8-60e8-43ee-9f66-9e813aa02e4e" />

**Comment by akriese** [comment]
I used the checkpoint that got saved automatically as it had the lowest validation loss.
Here are the my wandb charts for the finetuning:
<img width="1084" height="726" alt="Image" src="https://github.com/user-attachments/assets/3ce5d3ab-f7c3-4447-93c7-16c0f8e3a6d0" />

This might matter: I only ran the pretraining for 30 epochs due to time-constraints on my institute's cluster. These are the metrics for the run:
<img width="1093" height="724" alt="Image" src="https://github.com/user-attachments/assets/199a144a-50b8-4eaa-95a6-af9c5ddefd23" />

Are they similar to your pretraining stats, or wasnt my run able to converge and thus is a worse base for the finetuning?

**Comment by MichaelWangGo** [comment]
> Here are the my wandb charts for the finetuning: <img alt="Image" width="1084" height="726" src="https://private-user-images.githubusercontent.com/62887033/516368615-3ce5d3ab-f7c3-4447-93c7-16c0f8e3a6d0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM1Njg0MTcsIm5iZiI6MTc2MzU2ODExNywicGF0aCI6Ii82Mjg4NzAzMy81MTYzNjg2MTUtM2NlNWQzYWItZjdjMy00NDQ3LTkzYzctMTZjMGY4ZTNhNmQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE5VDE2MDE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2MDNmZDdhODZlNDM5ZWUwYTIwN2U2ZDcxMDE5Nzc2YWRjZGI3MjgxMzhmY2RhNmIzMDE1NDdiODljZTY0MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oPIg968hkFtD-untF1jbnddHekzoprmT1FKDjQDIirs">
> 
> This might matter: I only ran the pretraining for 30 epochs due to time-constraints on my institute's cluster. These are the metrics for the run: <img alt="Image" width="1093" height="724" src="https://private-user-images.githubusercontent.com/62887033/516370235-199a144a-50b8-4eaa-95a6-af9c5ddefd23.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM1Njg0MTcsIm5iZiI6MTc2MzU2ODExNywicGF0aCI6Ii82Mjg4NzAzMy81MTYzNzAyMzUtMTk5YTE0NGEtNTBiOC00ZWFhLTk1YTYtYWY5YzVkZGVmZDIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE5VDE2MDE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4YmY5OTVmNjdjNDRiM2ZhNTVhZDhmZmJkODkxN2U4MTdlNDJkNTUxODUyMzNlYWIzZTI2OGZmMDUzZmY3NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0riFSp2O2yMLLg2p1AgTASVWQhSDiWPSLHc1eXETiNM">
> 
> Are they similar to your pretraining stats, or wasnt my run able to converge and thus is a worse base for the finetuning?

Yes, I believe your model has room for convergence. A good pretraining really matters.
Here is my pretraining record:

<img width="1018" height="714" alt="Image" src="https://github.com/user-attachments/assets/a075716f-974e-4819-a4d3-6023a4d15d02" />
